Tests/FoundationEssentialsTests/URLTests.swift

```diff
@@ -1327,6 +1327,19 @@ final class URLTests : XCTestCase {
         comp = try XCTUnwrap(URLComponents(string: legalURLString))
         XCTAssertEqual(comp.string, legalURLString)
         XCTAssertEqual(comp.percentEncodedPath, colonFirstPath)
+
+        // Colons should be percent-encoded by URLComponents.string if
+        // they could be misinterpreted as a scheme separator.
+
+        comp = URLComponents()
+        comp.percentEncodedPath = "not%20a%20scheme:"
+        XCTAssertEqual(comp.string, "not%20a%20scheme%3A")
+
+        // These would fail if we did not percent-encode the colon.
+        // .string should always produce a valid URL string, or nil.
+
+        XCTAssertNotNil(URL(string: comp.string!))
```

Comment:
For future reference: we prefer using `XCTUnwrap` to unwrap the string (like you did above on line 1327). I'd also appreciate if we can use `XCTAssertEqual` to assert the exact content instead of less-informational `XCTAssertNotNil`